rbac/internal/views.py

```diff
@@ -338,6 +341,155 @@ def get_org_admin(request, org_or_account):
     return HttpResponse('Invalid method, only "GET" is allowed.', status=405)
 
 
+def user_lookup(request):
+    """Get all groups, roles, and permissions for a provided user via username or email.
+
+    If both params are provided, email is ignored and username is used.
+
+    GET /_private/api/utils/user_lookup/?username=foo&email=bar@redhat.com
+    """
+    if request.method != "GET":
+        return handle_error("Invalid http method - only 'GET' is allowed", 405)
+
+    username = request.GET.get("username")
+    email = request.GET.get("email")
+
+    try:
+        validate_user_lookup_input(username, email)
+    except ValueError as err:
+        return handle_error(f"Invalid request input - {err}", 400)
+
+    try:
+        user = get_user_from_bop(username, email)
+    except UserNotFoundError as err:
+        return handle_error(f"Not found - {err}", 404)
+    except Exception as err:
+        return handle_error(f"Internal error - couldn't get user from bop: {err}", 500)
+
+    result = {
+        "username": user["username"],
+        "email_address": user["email"],
+    }
+
+    try:
+        principals = Principal.objects.filter(username__iexact=user["username"])
+    except Exception as err:
+        logger.warning(f"error querying for principal with username: '{username}' in rbac, err: {err}")
+        return handle_error(f"Internal error - failed to query rbac for user '{username}'", 500)
+
+    if len(principals) > 1:
+        logger.warning(f"multiple principles with username '{username}' exist in rbac")
+        return handle_error(f"Internal error - multiple principles with username '{username}' exist in rbac", 500)
+
+    groups = Group.platform_default_set()
```

Comment:
Has to filter by tenant(org) to get the exact default group for that user. Would be great to have a unit test with custom default group

Reply:
oooo good to know ty! I'll add a test for coverage. I think it will be covered when I address your comment around getting orgId/tenant since we are getting the user from bop. I'll be sure to include test it out